packages/squiggle-lang/src/rescript/FR/FR_Danger_Aggregate.res

```diff
@@ -0,0 +1,171 @@
+open FunctionRegistry_Core
+open FunctionRegistry_Helpers
+
+let nameSpace = "Danger_Aggregate"
+let requiresNamespace = true
+
+module AggregateFs = {
+  module Helpers = {
+    let checker = (fn, xs, minLength) => {
+      switch E.A.length(xs) < minLength {
+      | true =>
+        Error(
+          "Aggregation method does not make sense with fewer than" ++
+          Belt.Int.toString(minLength) ++ " elements",
+        )
+      | false => {
+          let checkedIndividualXs = E.A.fmap(x =>
+            switch (x, x > 1.0, x < 0.0) {
+            | (0.0, _, _) => Error("0 is not a probability")
+            | (1.0, _, _) => Error("1 is not a probability")
+            | (_, true, _) => Error("Probabilities can't be higher than 1")
+            | (_, _, true) => Error("Probabilities can't be lower than 0")
+            | (_, false, false) => Ok(x)
+            }
+          , xs)
+          let checkedCollectiveXs = E.A.R.firstErrorOrOpen(checkedIndividualXs)
+          let result = switch checkedCollectiveXs {
```

Comment:
could be done with an fmap, I assume. checkedCollectiveXs |> E.R.fmap(fn)

Reply:
Not sure, firstErrorOrOpen translates `array<Ok<t>|Error<string>>` into `Ok<array<t>> | Error(string)`. So my sense is that it wouldn't.